index.js

```diff
@@ -3,30 +3,45 @@ const data = require('./data.js');
 // Returns an array of titles that are not in English
 // field: language
 function getAllNonEnglishTitles() {
-
+  const nonEnglish = data.filter(movie => movie.language !== 'English');
+  return nonEnglish;
 }
 
 // Returns an array with title names of titles
 // that lasts more than 2 hours and a half.
 // field: duration
 function titlesWithMoreThan2HoursAndAHalf() {
-
+  const moreThan = data.filter(movie => movie.duration > 150);
+  return moreThan;
 }
 
 // Returns an array of all Countries that have produced titles.
 // field: country
 function allCountries() {
-
+  const countries = [];
+  data.map((movie) => {
+    if (!countries.includes(movie.country)) countries.push(movie.country);
+    return null;
+  });
+  return countries;
 }
```

Comment:
It is working, but not performing. What can be done in order to improve performance?